kgforge/specializations/resolvers/agent_resolver.py

```diff
@@ -52,9 +53,11 @@ def _resolve(self, text: Union[str, List[str]], target: Optional[str], type: Opt
             family_name_filter = f" FILTER (?familyName = \"{text}\")"
             limit = 1
         elif strategy == strategy.EXACT_CASEINSENSITIVE_MATCH:
-            name_filter = f" FILTER regex(?name, \"^{text}$\", \"i\")"
-            given_name_filter = f" FILTER regex(?givenName, \"^{text}$\", \"i\")"
-            family_name_filter = f" FILTER regex(?familyName, \"^{text}$\", \"i\")"
+            tmp_text = re.sub(r'[-()\"#/@;:<>{}`+=~|.!?,]', "\\\\S", text)
```

Comment:
Here I keep track of the positions of the special characters (assigning a temporal value that I can replace), so in the next line I use `\p{Punct}` in those positions, to pass to the regex filter.